posts/hello-microservice-deployment-part-2.md

```diff
@@ -0,0 +1,358 @@
+
+Ok, so what is this container orchestration thing about? Let's revisit Webflix for this one. Let's say Webflix has gone down the microservices path (for their organization it is a pretty good course of action) and they have isolated a few aspect of their larger application into individual services. Each of those services are then built as images and need to be deployed in the form of live containers.
+
+![Webflix Services](../images/hello_micro_deploy/deploy-micro-orchestrated-services.png)
+
+The diagram here shows just a small portion of the Webflix application. Let's assume it's peak watching time, the little numbers next to the various services are examples of the number of live containers in play. In theory you would be able to make this happen through use of the `docker run` command.
+
+Now let's say one of the recommend_service containers hits an unrecoverable error and falls over. If we were managing this through `docker run` then we would need to notice this and then restart the offending container. Or suddenly we are no longer in peak time and the numbers of containers needs to be scaled down. Or you want to deploy a new version of the recommend_service image? Then there are other concerns like container startup time and aliveness probes. And what if you have a bunch of machines that you want to run your containers on, you would need to keep track of what resources are available on each machine, and what replicas are running on each machine so you can scale up and scale down when you need to.
+
+That is where container orchestration comes in.
+
+Please note, the author of this post is terribly biased and thinks K8s is the shizzle. So that is what we will cover here. There are alternative platforms that you might want to look into, such as Docker Swarm and Mesos/Marathon.
+
+Ok, back to the K8s goodness.
+
+K8s was born and raised and battle hardened at Google. Google used it internally to run huge production workloads, and then they open-sourced it because they are lovely people. So K8s is a freely available, highly scalable, extendable and generally solid platform. Another thing that is really useful is that Google cloud has first class K8s support, so if you have a K8s cluster on Google cloud then there are all sorts of tools available that make managing your application easier.
+
+K8s defines a bunch of concepts, there are entire books written on the subject so what follows is a surface level introduction. Personally I've found [this book](https://www.amazon.com/gp/product/1491935677/ref=as_li_tl?ie=UTF8&camp=1789&creative=9325&creativeASIN=1491935677&linkCode=as2&tag=sheena0d-20&linkId=e5bf88f134212c4beba73c601598d7ff") seriously useful.
```

Comment:
Lines 17 --> 33 can be made a lot more succinct. It's over 400 words, but I don't think it gives the reader anything actionable to take away with them.

Reply:
not sure about this: I think it's good to outline the problem k8s solves, say why k8s is solid, and point out alternatives.